crates/fuel-core/src/service/sub_services/algorithm_updater.rs

```diff
@@ -189,18 +190,22 @@ where
     }
 }
 
-pub fn get_synced_gas_price_updater<L2DataStore, L2DataStoreView>(
+pub fn get_synced_gas_price_updater<L2DataStore, L2DataStoreView, GasPriceStore>(
     config: Config,
     genesis_block_height: BlockHeight,
     settings: ConsensusParametersProvider,
-    mut gas_price_db: Database<GasPriceDatabase, RegularStage<GasPriceDatabase>>,
+    mut gas_price_db: GasPriceStore,
     on_chain_db: &L2DataStoreView,
     block_stream: BoxStream<SharedImportResult>,
     da_block_costs: DaBlockCostsSharedState,
-) -> anyhow::Result<Updater>
+) -> anyhow::Result<Updater<GasPriceStore>>
 where
     L2DataStore: L2Data,
     L2DataStoreView: AtomicView<LatestView = L2DataStore>,
+    GasPriceStore: GasPriceData
+        + MetadataStorage
+        + KeyValueInspect<Column = GasPriceColumn>
+        + Modifiable,
```

Comment:
We should be careful about these all being bound together. Obviously this is still in an intermediate state, so maybe goes without saying, but ultimately the different ports should be handled independently. If they are the same thing under the hood, that shouldn't be leaked to like this.

Reply:
I also don't know the reason for making it generic in this PR; I think we can do it later when we switch from the V0 to the V1 algorithm.